src/main/java/lovespiritual/lovespiritual.java

```diff
@@ -0,0 +1,356 @@
+                    markTask(input, taskCount, tasks);
+                    saveTasks();
+                } else if (input.startsWith("unmark")) {
+                    unmarkTask(input, taskCount, tasks);
+                    saveTasks();
+                } else if (input.startsWith("todo")) {
+                    taskCount = todo(input, tasks);
+                    saveTasks();
+                } else if (input.startsWith("deadline")) {
+                    taskCount = deadline(input, tasks);
+                    saveTasks();
+                } else if (input.startsWith("event")) {
+                    taskCount = event(input, tasks);
+                    saveTasks();
+                } else if (input.startsWith("delete")) {
+                    deleteTask(input, tasks, taskCount);
+                    saveTasks();
+                }
+                else {
+                    throw new lovespiritualException("(^_^) Let's get started with a command!");
+                }
+            } catch (lovespiritualException e) {
+                System.out.println(SEPARATOR);
+                System.out.println(e.getMessage());
+                System.out.println(SEPARATOR);
+            } catch (Exception e) {
+                System.out.println(SEPARATOR);
+                System.out.println("Oh no! (＞﹏＜) Something went a little wrong...");
+                System.out.println(SEPARATOR);
+            }
+        }
+    }
+
+    private static void deleteTask(String input, ArrayList<Task> tasks, int taskCount) throws lovespiritualException {
+        String taskNumber = input.substring("delete".length()).trim();
+        if (taskNumber.isEmpty()) {
+            throw new lovespiritualException("Oopsie! (⊙_⊙) Please give me a valid number!");
+        }
+        int indexNumber;
+        try {
+            indexNumber = Integer.parseInt(taskNumber) - 1;
+        } catch (NumberFormatException e) {
+            throw new lovespiritualException("Hmm, that's not a number! (・_・;) Try again, please!");
+        }
+        Task removedTask = tasks.get(indexNumber);
+        if (indexNumber >= 0 && indexNumber < taskCount) {
+            tasks.remove(indexNumber);
+            System.out.println(SEPARATOR);
+            System.out.println("Got it! (◠‿◠) This task is removed!");
+            System.out.println(removedTask);
+            System.out.println(SEPARATOR);
+        } else {
+            throw new lovespiritualException("Yikes! (≧Д≦) That number doesn't look right. Can you double-check it?");
+        }
+    }
+
+    private static void saveTasks() {
+        try {
+            File file = new File(FILE_PATH);
+            file.getParentFile().mkdirs();
+            BufferedWriter writer = new BufferedWriter(new FileWriter(FILE_PATH));
+            for (Task task : tasks) {
+                writer.write(savedFormat(task));
+                writer.newLine();
+            }
+            writer.close();
+        } catch (Exception e) {
```

Comment:
You should handle exceptions more specifically to avoid masking actual issues and making debugging harder. Consider catching IOException separately for file operations and only catch Exception for truly unexpected issues.